### ISSUE TITLE
vine: submit task and expire waiting both rotate the ready list of tasks

### ISSUE THREAD

**Issue by btovar** [cooperative-computing-lab/cctools#3696]
Almost never an issue, but it allows for a pathological case:

Assume n is q->attempt_schedule_depth, and there are 2n tasks in the ready list. The first n are too large for the current workers, and the rest could be scheduled, then:

1. Manager tries to schedule first n tasks. They are to large to fit in any worker.
2. Manager tries to expire second n tasks. None of them are expired.
3. Repeat...

One easy way to solve this would be to change `expire_waiting_tasks` to `expire_waiting_task` and make it part of `send_one_task`.

@colinthomas-z80?

**Comment by colinthomas-z80** [comment]
Yes that is tricky... Would we ever expire the tasks that don't fit? Or would this just prevent this situation with the 2n task groups
